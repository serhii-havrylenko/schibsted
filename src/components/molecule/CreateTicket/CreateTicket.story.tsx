import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import CreateTicket, { CreateTicketProps } from './CreateTicket';

const props: CreateTicketProps = {
  onCreateClick: action('create clicked'),
  titleRef: React.createRef() as React.RefObject<HTMLInputElement>,
  descriptionRef: React.createRef() as React.RefObject<HTMLInputElement>,
};

storiesOf('CreateTicket', module).add('default', () => (
  <CreateTicket {...props} />
));
