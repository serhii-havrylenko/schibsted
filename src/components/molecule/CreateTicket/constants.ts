export const RequiredError = 'This field is required';
