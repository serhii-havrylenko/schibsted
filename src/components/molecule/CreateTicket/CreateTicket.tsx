import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import TextField from '@material-ui/core/TextField';
import * as React from 'react';

import Button from '#atom/Button';
import CardContentWrapper from '#atom/CardContentWrapper';
import { RequiredError } from './constants';

export interface CreateTicketProps {
  titleRef: React.RefObject<HTMLInputElement>;
  descriptionRef: React.RefObject<HTMLInputElement>;
  titleError?: boolean;
  descriptionError?: boolean;
  onCreateClick: () => void;
}

const CreateTicket: React.SFC<CreateTicketProps> = ({
  onCreateClick,
  titleRef,
  titleError,
  descriptionRef,
  descriptionError,
}) => (
  <Card>
    <CardContentWrapper>
      <TextField
        margin="dense"
        inputRef={titleRef}
        label="Title"
        required={true}
        type="text"
        error={titleError}
        helperText={titleError && RequiredError}
      />
      <TextField
        margin="dense"
        inputRef={descriptionRef}
        label="Description"
        multiline={true}
        required={true}
        rows={5}
        type="text"
        error={descriptionError}
        helperText={descriptionError && RequiredError}
      />
    </CardContentWrapper>
    <CardActions>
      <Button onClick={onCreateClick}>Create</Button>
    </CardActions>
  </Card>
);

export default CreateTicket;
