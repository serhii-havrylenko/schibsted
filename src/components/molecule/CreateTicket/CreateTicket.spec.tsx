import { shallow } from 'enzyme';
import * as React from 'react';

import CreateTicket, { CreateTicketProps } from './CreateTicket';

describe('<CreateTicket /> tests', () => {
  const props: CreateTicketProps = {
    onCreateClick: jest.fn(),
    titleRef: React.createRef() as React.RefObject<HTMLInputElement>,
    descriptionRef: React.createRef() as React.RefObject<HTMLInputElement>,
  };

  test('should match snapshot without errors', () => {
    const tree = shallow(<CreateTicket {...props} />);
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot with error in title', () => {
    const tree = shallow(<CreateTicket {...props} titleError={true} />);
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot with error in description', () => {
    const tree = shallow(<CreateTicket {...props} descriptionError={true} />);
    expect(tree).toMatchSnapshot();
  });
});
