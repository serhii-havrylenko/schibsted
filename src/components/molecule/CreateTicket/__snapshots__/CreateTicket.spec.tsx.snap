// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTicket /> tests should match snapshot with error in description 1`] = `
<WithStyles(Card)>
  <Styled(WithStyles(CardContent))>
    <TextField
      inputRef={
        Object {
          "current": null,
        }
      }
      label="Title"
      margin="dense"
      required={true}
      select={false}
      type="text"
      variant="standard"
    />
    <TextField
      error={true}
      helperText="This field is required"
      inputRef={
        Object {
          "current": null,
        }
      }
      label="Description"
      margin="dense"
      multiline={true}
      required={true}
      rows={5}
      select={false}
      type="text"
      variant="standard"
    />
  </Styled(WithStyles(CardContent))>
  <WithStyles(CardActions)>
    <Button
      onClick={[MockFunction]}
    >
      Create
    </Button>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;

exports[`<CreateTicket /> tests should match snapshot with error in title 1`] = `
<WithStyles(Card)>
  <Styled(WithStyles(CardContent))>
    <TextField
      error={true}
      helperText="This field is required"
      inputRef={
        Object {
          "current": null,
        }
      }
      label="Title"
      margin="dense"
      required={true}
      select={false}
      type="text"
      variant="standard"
    />
    <TextField
      inputRef={
        Object {
          "current": null,
        }
      }
      label="Description"
      margin="dense"
      multiline={true}
      required={true}
      rows={5}
      select={false}
      type="text"
      variant="standard"
    />
  </Styled(WithStyles(CardContent))>
  <WithStyles(CardActions)>
    <Button
      onClick={[MockFunction]}
    >
      Create
    </Button>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;

exports[`<CreateTicket /> tests should match snapshot without errors 1`] = `
<WithStyles(Card)>
  <Styled(WithStyles(CardContent))>
    <TextField
      inputRef={
        Object {
          "current": null,
        }
      }
      label="Title"
      margin="dense"
      required={true}
      select={false}
      type="text"
      variant="standard"
    />
    <TextField
      inputRef={
        Object {
          "current": null,
        }
      }
      label="Description"
      margin="dense"
      multiline={true}
      required={true}
      rows={5}
      select={false}
      type="text"
      variant="standard"
    />
  </Styled(WithStyles(CardContent))>
  <WithStyles(CardActions)>
    <Button
      onClick={[MockFunction]}
    >
      Create
    </Button>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
